pages/case-study/[id].tsx

```diff
@@ -0,0 +1,30 @@
+import { CaseStudyPage } from "../components/caseStudyInnerPage/CaseStudyInnerPage";
+import { Header } from "../components/header/Header";
+import { Footer } from "../components/footer/Footer";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import { httpClient } from "../configs/api/httpClient/HttpClient";
+
+const CaseStudy = () => {
+  let { query: { id } } = useRouter();
+  const [ caseStudy, setCaseStudy ] = useState(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
```

Comment:
```suggestion
    const fetchData = async () => {
```

I think we can create general reusable fetchData function

Reply:
we can't, because fetchData function do different functionality every time, except calling httpClient function inside of it